code/lab/lab.cpp

```diff
@@ -1323,15 +1333,20 @@ void labviewer_make_ship_window(Button * /*caller*/)
 		Lab_species_nodes = NULL;
 	}
 
+	ship_list_endpoints = new std::pair<TreeItem*, TreeItem*>[Species_info.size()];
```

Comment:
This should use a vector instead. There is no need to use dynamic allocation here.